typescript/schedules.md

```diff
@@ -0,0 +1,1257 @@
+  action: {
+    workflowId: 'wf-biz-id',
+    type: myWorkflow,
+    args: ['arg1', 'arg2'],
+  },
+  overlap: ScheduleOverlapPolicy.BufferOne,
+  catchupWindow: '2m',
+  pauseOnFailure: true,
+  note: 'started schedule',
+  pause: true, // this sets `state.paused: true` (default false)
+  limitedActions: 10,
+  triggerImmediately: true,
+  backfill: [
+    {
+      start: new Date(),
+      end: new Date(),
+      overlap: ScheduleOverlapPolicy.AllowAll,
+    },
+  ],
+  memo,
+  searchAttributes,
+});
+
+const schedule = client.getHandle('schedule-biz-id');
+
+const scheduleDescription = await schedule.describe();
+
+// For later, pending API finalization:
+// https://github.com/temporalio/proposals/pull/62/files#r933532170
+// const matchingStartTimes = await schedule.listMatchingTimes({ start: new Date(), end: new Date() })
+// const matchingStartTimes = await client.listMatchingTimes({ spec, start, end })
+
+await schedule.update(
+  (schedule) => {
+    schedule.spec.intervals[0].every = '1d';
+    // unset with:
+    // delete schedule.spec.intervals;
+
+    // return false to stop retrying early
+  },
+  { retry: retryPolicy }
+);
+
+await schedule.trigger();
+await schedule.backfill({ startAt: new Date(), endAt: addWeeks(new Date(), 1) }); // also takes array
+await schedule.pause('note: pausing');
+await schedule.unpause('now unpause');
+
+await schedule.delete();
+
+const { schedules, nextPageToken } = await client.listByPage({
+  pageSize: 50,
+  nextPageToken: 'string',
+});
+
+for await (const schedule: Schedule of client.list()) {
+  const { id, memo, searchAttributes } = schedule;
+  // ...
+}
+```
+
+### Types
+
+```ts
+import { DataConverter, LoadedDataConverter } from '@temporalio/common';
+import { loadDataConverter } from '@temporalio/internal-non-workflow-common';
+import {
+  composeInterceptors,
+  Headers,
+  Replace,
+  RetryPolicy,
+  SearchAttributes,
+  Workflow,
+} from '@temporalio/internal-workflow-common';
+import { temporal } from '@temporalio/proto';
+import os from 'os';
+import { Connection } from './connection';
+import { ScheduleClientCallsInterceptor, ScheduleClientInterceptors } from './interceptors';
+import { ConnectionLike, Metadata, WorkflowService } from './types';
+import { WorkflowHandle, WorkflowStartOptions } from './workflow-client';
+
+import type { NonNegativeInteger, RequireAtLeastOne } from 'type-fest';
+
+// TODO is a non-generic NonNegativeInteger possible? The ones below error due to no type argument
+export type PositiveInteger = Exclude<NonNegativeInteger, 0>;
+
+/**
+ * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+ */
+export type Milliseconds = string | NonNegativeInteger;
+
+/**
+ * @format number of seconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+ */
+export type Seconds = string | NonNegativeInteger;
+
+export interface UpdateScheduleOptions {
+  /**
+   * @default TODO
+   */
+  retry?: RetryPolicy;
```

Comment:
This is overkill for client side retry policy IMO, I'd just retry up to X times with no delay.
It's only for the optimistic update.

Reply:
Default 10? 

<img width="1412" alt="image" src="https://user-images.githubusercontent.com/251288/184214930-6729dc63-a50b-440f-af8f-1a2e9d3f369d.png">

Reply:
Don't know what a good default would be here, 2? 3?

Reply:
Using 3